presubmit.sh

```diff
@@ -1,92 +1,58 @@
 #!/bin/bash
 
-set -e
+set -Eeuo pipefail
 
 git init
 
 # Stage files to track their history
 git add .
 
 git remote add origin "https://github.com/GoogleCloudPlatform/dataproc-initialization-actions.git"
-git fetch origin
+git fetch origin master
 
 # Infer the files that changed
-CHANGED_FILES=$(git diff origin/master --name-only)
-echo "Files changed: $CHANGED_FILES"
+mapfile -t CHANGED_FILES < <(git diff origin/master --name-only)
+echo "Changed files: ${CHANGED_FILES[*]}"
 
 # Determines whether a given string is a substring of any changed file name
 contains() {
-	for file in $CHANGED_FILES
-	do
-	    if [[ $file =~ $1 ]]; then
-	    	return 0
-		fi
-	done
-	return 1
+  for file in "${CHANGED_FILES[@]}"
+  do
+    if [[ $file =~ $1 ]]; then
+      return 0
+    fi
+  done
+  return 1
 }
 
-# Determines whether a given string is the prefix of any changed file name
-contains_prefix() {
-	prefix="${1}*"
-	for file in $CHANGED_FILES
-	do
-	    if [[ $file == $prefix ]]; then
-	    	return 0
-		fi
-	done
-	return 1
-}
+declare -a DIRECTORIES_TO_TEST
+for dir in */
+do
+  # skip not init action changes
+  if [[ $dir =~ ^(integration_tests/|util/)$ ]]; then
+    continue
+  fi
+  if contains "$dir"; then
```

Comment:
Nice shell script wizardry.

Does this account for the edge case where, for instance, starburst-presto is modified and presto is not? In this case, only starburst-presto/test_presto.py should be triggered.

Reply:
Good catch, modified patter to match prefix from the beginning of the string instead of contains.